open_lm/positional_embedding/rotary.py

```diff
@@ -41,42 +45,43 @@ class RotaryEmbedding(torch.nn.Module):
         (it does not create the embedding dimension) and will likely be picked up (imported) on a ad-hoc basis
     """
 
-    def __init__(self, dim_model: int, *_, **__):
+    def __init__(self, dim_model: int, seq_len: int, *_, **__):
         super().__init__()
         # Generate and save the inverse frequency buffer (non trainable)
         inv_freq = 1.0 / (10000 ** (torch.arange(0, dim_model, 2).float() / dim_model))
         self.register_buffer("inv_freq", inv_freq)
 
-        self._seq_len_cached = None
         self._cos_cached = None
         self._sin_cached = None
+        self._seq_len_cached = 0
+        self._update_cos_sin_tables(seq_len)
 
-    def _update_cos_sin_tables(self, x, seq_dimension=1):
-        seq_len = x.shape[seq_dimension]
-
+    def _update_cos_sin_tables(self, seq_len: int = None, device: torch.device = None, dtype: torch.dtype = None):
         # Reset the tables if the sequence length has changed,
```

Comment:
Is this comment outdated now?

Reply:
Also, might be worth leaving a comment explaining when seq_len can be `> self.seq_len_cached`. I assume this happens when we're generating as the sequence gets longer, but worth dropping a comment imo.

Reply:
Good catch and fair point, I’ll update that.
You mean explaining the assert statement right?

Reply:
hm, I'm not seeing the assert - I meant the two if statements:
`        if seq_len is None or seq_len < self._seq_len_cached:`
and
`        if seq_len > self._seq_len_cached or self._cos_cached.device != device or self._cos_cached.dtype != dtype:`